controllers/cart.js

```diff
@@ -0,0 +1,64 @@
+const { products, carts, orders } = require('../storage');
+const crypto = require('crypto');
+
+function renderCart(id) {
+  if (carts.find((item) => item.userId === id)) {
+    return;
+  } else {
+    const cart = {
+      id: crypto.randomUUID(),
+      userId: id,
+      products: [],
+    };
+    carts.push(cart);
+  }
+}
+
+function addProductToCart(id, product) {
+  const thisProduct = products.find((item) => item.id === +product);
+  const thisCart = carts.find((item) => item.userId === id);
+  const arrOfProducts = thisCart.products;
+  arrOfProducts.push(thisProduct);
+  thisCart.products = arrOfProducts;
+}
+
+const updateCart = (req, res) => {
+  const headerUserId = req.headers['x-user-id'];
+  const { productId } = req.params;
+  renderCart(headerUserId);
+  addProductToCart(headerUserId, productId);
+
+  res.status(200).send(carts);
+};
+
+const deleteProductFromCart = (req, res) => {
+  const headerUserId = req.headers['x-user-id'];
+  const { productId } = req.params;
+  const thisCart = carts.find((item) => item.userId === headerUserId);
+  const arrOfItems = thisCart.products;
+  thisCart.products = arrOfItems.filter((item) => item.id !== +productId);
```

Comment:
`thisCart.products.filter((item) => item.id !== +productId);` 
It's not necessary to create `arrOfItems ` variable.